streams/src/main/java/org/apache/kafka/streams/errors/DeserializationExceptionHandler.java

```diff
@@ -37,11 +37,27 @@ public interface DeserializationExceptionHandler extends Configurable {
      * @param context processor context
      * @param record record that failed deserialization
      * @param exception the actual exception
+     * @deprecated Please {@link #handle(ErrorHandlerContext, ConsumerRecord, Exception)}
+     */
+    @Deprecated
+    default DeserializationHandlerResponse handle(final ProcessorContext context,
+                                                  final ConsumerRecord<byte[], byte[]> record,
+                                                  final Exception exception) {
+        throw new UnsupportedOperationException();
```

Comment:
Nit. KIP says `NotImplementedException()` -- we should update the KIP accordingly.

Reply:
done KIP has been updated